TESTING.md

```diff
@@ -0,0 +1,59 @@
+# Testing 'oneview' cookbook
+The 'oneview' cookbook provides already unit testing strategy, besides having linting tools.
+This document will cover on how to execute and implement these tests.
+
+## Executing tests
+The tests can be executed by `rake` tasks, `guard` watches, and by their original tools, like `rspec`, `chefspec`, `rubocop` and `foodcritic`.
+
+### Rake
+All the test strategies and checks can be executed by the rake command.
+Please use `rake -T` to see a full list of commands.
+
+### Guard
+Guard will watch the changes and execute unit tests and style checks. Just activate it to have real time test execution on every change.
+
+### Unit tests
+All unit tests are inside spec folder. You can execute them manually by using the RSpec and specifying the test files, like `rspec spec/path/to/my/tests`.
+
+For pure lib tests the projects use solely RSpec, however, for Chef custom resources tests we use a variation of RSpec, the ChefSpec. Both of them work the same way by issuing the `rspec` command.
+
+#### Coverage
+The unit tests issue a coverage report generated by SimpleCov for the pure lib files, and ChefSpec issues the report for the Chef custom resources (Since SimpleCov does not support it).
+
+However, you must pay attention because SimpleCov's and ChefSpec's coverage report conflict with each other, i.e. as soon ChefSpec's coverage starts running, it will interrupt the SimpleCov's coverage status and it will generate the report at that point. So, because of this, is extremely important running all pure ruby lib files **before** any Chef custom resource, otherwise, the SimpleCov report would be **incorrect**.
+
+The main difference between SimpleCov and ChefSpec coverage is that SimpleCov reports line coverage and ChefSpec shows custom resource touch coverage.
+
+### Style checking
+Style checking is done by `rubocop` for all the custom resource providers and pure ruby lib files, and `foodcritic` to analyze testing recipes and examples.
+
+### Functional tests
```

Comment:
Every resource must have an example for each action that shows how to use it. These examples can be used to do functional and OneView integration testing but they need to be executed manually using ~the~ Chef.